src/peep/peep.c

```diff
@@ -3421,7 +3457,511 @@ static void peep_update_fixing(int steps, rct_peep* peep){
 		return;
 	}
 
-	RCT2_CALLPROC_X(RCT2_ADDRESS(0x006C0EB0, uint32)[peep->sub_state], steps, 0, 0, 0, (int)peep, (int)ride, 0);
+	switch (peep->sub_state) {
+		case 0:
+			loc_6C0EEC(steps, peep, ride);
+			break;
+
+		case 1:
+			loc_6C0FAA(steps, peep, ride);
+			break;
+
+		case 2:
+		case 3:
+		case 4:
+		case 5:
+			loc_6C100D(steps, peep, ride);
+			break;
+
+		case 6:
+			loc_6C10A5(steps, peep, ride);
+			break;
+
+		case 7:
+			loc_6C11CC(steps, peep, ride);
+			break;
+
+		case 8:
+			loc_6C121F(steps, peep, ride);
+			break;
+
+		case 9:
+			loc_6C133F(steps, peep, ride);
+			break;
+
+		case 10:
+			loc_6C13B4(steps, peep, ride);
+			break;
+
+		case 11:
+			loc_6C13F8(steps, peep, ride);
+			break;
+
+		case 12:
+			loc_6C14E5(steps, peep, ride);
+			break;
+
+		case 13:
+			loc_6C1559(steps, peep, ride);
+			break;
+
+		case 14:
+			loc_6C15EF(steps, peep, ride);
+			break;
+
+		default:
+			log_error("Invalid substate");
+			RCT2_CALLPROC_X(RCT2_ADDRESS(0x006C0EB0, uint32)[peep->sub_state], steps, 0, 0, 0, (int) peep, (int) ride, 0);
+	}
+
+}
+
+void loc_6C0EEC(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	ride->mechanic_status = 3;
+	ride->window_invalidate_flags |= RIDE_INVALIDATE_RIDE_MAINTENANCE;
+
+	loc_6C1653(steps, peep, ride);
+}
+
+void loc_6C0F09(int steps, rct_peep *peep, rct_ride *ride) {
+	uint32 ebx = ride->broken_vehicle;
+	uint16 bp = ride->vehicles[peepDirection];
+
+	if (bp == 0xFFFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	rct_sprite *ebp_sprite = &g_sprite_list[bp];
+	sint8 ah = ride->broken_car;
+
+	// loc_6C0F3E:
+	while (true) {
```

Comment:
Maybe this could be a for loop?

    for (sint8 ah = ride->broken_car; ah > 0; --ah)